language/move-lang/tests/move_check_testsuite.rs

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use move_lang::{move_check_no_report, shared::Address};
+use std::{fs, path::Path};
+use text_diff;
+
+use move_lang::test_utils::*;
+
+const OUT_EXT: &str = "out";
+const EXP_EXT: &str = "exp";
+
+const KEEP_TMP: &str = "KEEP";
+const UPDATE_BASELINE: &str = "UPDATE_BASELINE";
+
+fn format_diff(expected: String, actual: String) -> String {
+    use text_diff::*;
+    let (_, changeset) = diff(&expected, &actual, "\n");
+    let mut ret = String::new();
+
+    for seq in changeset {
+        match &seq {
+            Difference::Same(x) => {
+                ret.push_str(x);
+                ret.push_str("\n");
+            },
+            Difference::Add(x) => {
+                ret.push_str("\x1B[92m");
+                ret.push_str(x);
+                ret.push_str("\x1B[0m");
+                ret.push_str("\n");
+            },
+            Difference::Rem(x) => {
+                ret.push_str("\x1B[91m");
+                ret.push_str(x);
+                ret.push_str("\x1B[0m");
+                ret.push_str("\n");
+            }
+        }
+    }
+    ret
+}
+
+// Runs all tests under the test/testsuite directory.
+fn move_check_testsuite(path: &Path) -> datatest_stable::Result<()> {
+    let targets: Vec<&'static str> = vec![Box::leak(Box::new(path.to_str().unwrap().to_owned()))];
```

Comment:
I assume this is because `fn move_check` expects static str references, but could I ask why?

Reply:
I store them in each `Loc` which lets me have them implement `Copy` very easily. In other words, this is a really shitty way of interning the strings. We should probably build an intern table for all of the parsed/input strings eventually 

https://github.com/libra/libra/blob/master/language/move-lang/src/shared/mod.rs#L113#L117

Reply:
Ah I see, other wise you would have to spread lifetimes everywhere you use these. Yeah let's solve it later.

Reply:
The thing is always going to have a `'static` lifetime. For the interned strings, by design, you want them to live as long as the compiler is up and running. The shitty part is that I'm just calling Box::leak in an ad-hoc fashion instead of building an actual table